### PR TITLE
Add CoverageUnion wrapper to `geos` package

### PR DESCRIPTION
**Pull request by peterstace**
## Description

- Add CoverageUnion wrapper to `geos` package. This is useful for when you want to union together geometries that are noded together and don't overlap because it's much faster than regular union.

## Check List

Have you:

- Added unit tests? Yes.

- Add cmprefimpl tests? (if appropriate?) N/A

- Updated release notes? (if appropriate?): Yes.

## Related Issue

- https://github.com/peterstace/simplefeatures/issues/505

### CODE REVIEW

File: geos/entrypoints_test.go [hunk 1]
```diff
@@ -815,3 +815,41 @@ func TestMakeValid(t *testing.T) {
...
+		},
+		{
+			// Input constraint violated: inputs overlap.
+			input:   `GEOMETRYCOLLECTION(POLYGON((0 0,0 1,1 0,0 0)),POLYGON((0 0,0 1,1 1,0 0)))`,
+			wantErr: true,
+		},
+		{
+			// Input constraint violated: not noded correctly.
+			input:   `GEOMETRYCOLLECTION(POLYGON((0 0,0 1,1 0,0 0)),POLYGON((0 0,2 0,0 -2,0 0)))`,
+			wantErr: true,
```
Comment (peterstace) [inline geos/entrypoints_test.go:837]: Still need to confirm that the errors coming back contain reasonable content.

**Comment by peterstace** [comment]
I noticed a case where the error reporting isn't quite correct. It's due to what I believe may be a bug in GEOS itself (see https://github.com/libgeos/geos/issues/889 for details). I don't think it makes sense to work around the issue in the wrapper layer.

File: geos/entrypoints.go [hunk 2]
```diff
@@ -337,3 +347,29 @@ func MakeValid(g geom.Geometry, opts ...geom.ConstructorOption) (geom.Geometry,
...
+
+// The CoverageUnion function is used to union polygonal inputs that form a
+// coverage, which are typically provided as a GeometryCollection. This method
+// is much faster than other unioning methods, but there are some constraints
+// that must be met by the inputs to form a valid polygonal coverage. These
+// constraints are:
+//
+//  1. all input geometries must be polygonal,
+//  2. the inputs must not overlap, and
+//  3. the input boundaries must be noded in the same way.
```
Comment (albertteoh) [inline geos/entrypoints.go:359]: Does this sentence mean that the input boundaries share a common edge? Or is there a different meaning?

File: geos/entrypoints_test.go [hunk 3]
```diff
@@ -815,3 +815,63 @@ func TestMakeValid(t *testing.T) {
...
+		},
+		{
+			// Input constraint violated: not noded correctly.
+			input: `GEOMETRYCOLLECTION(
+				POLYGON((0 0,0 1,1 1,1 0,0 0)),
+				POLYGON((0 1,2 1,2 2,0 2,0 1))
+			)`,
+			wantErr: true,
+		},
+		{
```
Comment (albertteoh) [inline geos/entrypoints_test.go:856]: Would it be useful adding another test case where only the vertex is shared?

```
		{
			// Input constraint violated: not noded correctly - shared vertex but not edge.
			input: `GEOMETRYCOLLECTION(
				POLYGON((0 0,0 1,1 1,1 0,0 0)),
				POLYGON((0 1,2 1,2 2,1 2,0 1))
			)`,
			wantErr: true,
		},
```

I was also thinking of an example where polygons have no shared nodes (I hope that's the right terminology!); put simply, they're not touching. Although, maybe the tests already somewhat imply this.

Comment (peterstace) [reply geos/entrypoints_test.go:873]: It's actually a valid coverage if just a vertex is shared (which indicates my doc comment is probably not good enough!). I've added some more test cases in dccf76d65b282d8d2102e0b534bbd6daf550281e as well.

**Review by albertteoh — APPROVED** [review]
Looks good!

**Comment by peterstace** [comment]
Thanks for reviewing!

**Pull request merged**